src/components/MapView/Layers/BoundaryDropdown.tsx

```diff
@@ -0,0 +1,260 @@
+  // acts on the `value` prop, which we need to hide from <Select/> since this isn't a menu item.
+  return (
+    <FormControl {...rest}>
+      <InputLabel>{isMobile ? 'Tap' : 'Click'} the map to select</InputLabel>
+      <Select
+        multiple
+        onClose={() => {
+          // empty search so that component shows correct options
+          // otherwise, we would only show selected options which satisfy the search
+          setTimeout(() => setSearch(''), TIMEOUT_ANIMATION_DELAY);
+        }}
+        value={selectedBoundaries}
+        onChange={e => {
+          // do nothing if value is invalid
+          // This happens when you click list subheadings.
+          if (
+            !Array.isArray(e.target.value) ||
+            e.target.value.includes(undefined)
+          ) {
+            return;
+          }
+          dispatch(
+            setSelectedBoundaries(
+              Array.isArray(e.target.value) ? e.target.value : [],
+            ),
+          );
+        }}
+      >
+        <SearchField search={search} setSearch={setSearch} />
+        {!search && (
+          <MenuItem onClick={selectOrDeselectAll}>
+            {selectedBoundaries.length === 0 ? 'Select All' : 'Deselect All'}
+          </MenuItem>
+        )}
+        {search && categories.flatMap(c => c.children).length === 0 && (
+          <MenuItem disabled>No Results</MenuItem>
+        )}
+        {categories.reduce<ReactElement[]>(
+          // map wouldn't work here because <Select> doesn't support <Fragment> with keys, so we need one array
+          (components, category) => [
+            ...components,
+            <ClickableListSubheader
+              key={category.title}
+              onClick={e => {
+                e.preventDefault();
+                // if at least one is selected, deselect all. Otherwise select all
+                const categoryValues = category.children.map(c => c.value);
+                const selectedChildren =
+                  selectedBoundaries.filter(val => categoryValues.includes(val))
+                    .length > 0;
+                dispatch(
+                  setSelectedBoundaries(
+                    selectedChildren
+                      ? selectedBoundaries.filter(
+                          val => !categoryValues.includes(val),
+                        )
+                      : [...selectedBoundaries, ...categoryValues],
+                  ),
+                );
+              }}
+            >
+              <Typography variant="body2" color="primary">
+                {category.title}
+              </Typography>
+            </ClickableListSubheader>,
+            ...category.children.map(({ label, value }) => (
+              <MenuItem key={value} value={value}>
+                {label}
+              </MenuItem>
+            )),
+          ],
+          [],
+        )}
+      </Select>
+    </FormControl>
+  );
+}
+
+interface BoundaryDropdownProps {
```

Comment:
New prop ideas for @JorgeMartinezG to make this support single selection:
- `menuMode` if given makes the component act more like a dropdown menu than a selector. This can mean replacing the current Select parent with Menu (children should be the same)
- `onSelect` listener which could override default behavior (saving to redux state)

Or we could follow a HOC (higher order component) pattern where we have a super simple base component, and 2 extra components (one which provides multi select functionality and saves to redux state, and another which handles the goto feature). Might need more work however.

Reply:
Agree with second option. The base component should receive the event handler

Reply:
I did e7eea9159c884d458aaf392b5b3ec707d57907e7 to convert the component to HOC. You can extend on it in #329 for it to behave more like a menu ;)